server/routes/api.js

```diff
@@ -68,7 +68,7 @@ export default (storage) => {
 
   // Allow end users to authenticate.
   api.use(middlewares.authenticateUsers.optional({
-    domain: config('AUTH0_CUSTOM_DOMAIN'),
+    domain: config('IS_APPLIANCE') ? config('AUTH0_DOMAIN') : config('AUTH0_CUSTOM_DOMAIN'),
```

Comment:
@zxan1285 can you tell me anything about the trade-offs about making this a decision based on whether the extension is running on a PSaaS or Cloud? I think each time we make these two environments different we're taking a step back.

Reply:
Those environments are different, unfortunately. I'd really appreciate if PSaaS work exactly the same as the cloud, but currently it does not. Custom domains are working differently on those environments, so extensions have to adapt.

Reply:
Sounds good to me, thanks for clarifying.